src/gossip/data.zig

```diff
@@ -1018,7 +1017,9 @@ pub const ContactInfo = struct {
     const Self = @This();
 
     pub fn toNodeInstance(self: *Self) NodeInstance {
-        return NodeInstance.init(self.pubkey, @intCast(std.time.milliTimestamp()));
+        const prng_seed: u64 = @intCast(std.time.milliTimestamp());
+        var prng = std.Random.Xoshiro256.init(prng_seed);
+        return NodeInstance.init(prng.random(), self.pubkey, @intCast(std.time.milliTimestamp()));
     }
```

Comment:
sure, have just added a random interface parameter to all the functions requested in your review comments:
- ContactInfo.toNodeInstance()
- GossipPullFilterSet.initTest()
- GossipService.rotateActiveSet()

P.S: As mentioned in the PR description, a parameter is added to "all functions that are **directly** generating random numbers". This is because I wanted to be safe as I don't understand the entire codebase yet and thus where else it may be relevant to have this parameter.